Purchases/Public/RCPurchases.m

```diff
@@ -297,22 +307,17 @@ - (instancetype)initWithAppUserID:(nullable NSString *)appUserID
         self.operationDispatcher = operationDispatcher;
         self.introEligibilityCalculator = introEligibilityCalculator;
         self.receiptParser = receiptParser;
-
-        RCReceivePurchaserInfoBlock callDelegate = ^void(RCPurchaserInfo *info, NSError *error) {
-            if (info) {
-                [self sendUpdatedPurchaserInfoToDelegateIfChanged:info];
-            }
-        };
+        self.purchaserInfoManager = purchaserInfoManager;
 
         [self.identityManager configureWithAppUserID:appUserID];
 
         [self.systemInfo isApplicationBackgroundedWithCompletion:^(BOOL isBackgrounded) {
             if (!isBackgrounded) {
                 [self.operationDispatcher dispatchOnWorkerThreadWithRandomDelay:NO block:^{
-                    [self updateAllCachesWithCompletionBlock:callDelegate];
+                    [self updateAllCachesWithCompletionBlock:nil];
```

Comment:
the call delegate only called the purchaserInfoDelegate. 
the logic to call the purchaserInfoDelegate when purchaserInfo changes has been moved to the `purchaserInfoManager`, and centralized so that any time you update `purchaserInfo`, the delegate gets called, without the need for a separate completion block.